python/pyproject.toml

```diff
@@ -23,19 +23,18 @@ build-backend = "maturin"
 name = "hudi"
 description = "Native Python binding for Apache Hudi, based on hudi-rs."
 urls = { repository = "https://github.com/apache/hudi-rs/tree/main/python/" }
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 keywords = ["apachehudi", "hudi", "datalake", "arrow"]
 license = "Apache License 2.0"
 classifiers = [
     "License :: OSI Approved :: Apache Software License",
-    "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12"
 ]
 dependencies = [
-    "pyarrow>=8",
+    "pyarrow>=17",
     "pyarrow-hotfix",
```

Comment:
this is the minimum required version for pyarrow, isn't making it at least 17 too aggressive? also can you check if pyarrow-hotfix still needed if we upgrade to a newer version here

Reply:
pyarrow-hotfix is not needed since 0.14.1

I don't have a strong opinion on upgrading this, it seemed like a good idea, but I can revert this

Reply:
thanks for the useful info. what i meant was: this would make hudi-rs pyarrow requirement too strict and exclude people to use when they have lower pyarrow version. I'd align with datafusion's version requirement since we have the integration and it's not too old or too new

Reply:
sure, sounds good thanks